pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -343,6 +347,17 @@ func (p *Provisioner) initializeCreateOrGrant(options *apibkt.BucketOptions) err
 		return errors.Wrapf(err, "failed to set CA cert for the OBC %q to connect with object store %q via TLS", obc.Name, p.objectStoreName)
 	}
 
+	var httpClient *http.Client
+	if p.tlsCert != nil {
+		httpClient.Transport = object.BuildTransportTLS(p.tlsCert)
+	}
+	adminOpsClient, err := admin.New(p.endpoint, p.accessKeyID, p.secretAccessKey, httpClient)
```

Comment:
@leseb : Just wondering what this means for each request on Grant() or Provision() will result from the creation of adminsOpsClient() do we need really need that, instead can we use the same `adminopsclient` initialised as part of cephobjectstore and use that refer here.

Secondly `p.accessKeyID`, `p.secretAccesskey` specific to user of each OBC request which does not have capabilities to remove or create user. This is used to pass back to OBC request. For Provision() call, we may need to create bucket with this credentials otherwise we just give back details in OB response

Reply:
> @leseb : Just wondering what this means for each request on Grant() or Provision() will result from the creation of adminsOpsClient() do we need really need that? instead can we use the same `adminopsclient` initialised as part of cephobjectstore and use that refer here.


How?

> Secondly `p.accessKeyID`, `p.secretAccesskey` specific to user of each OBC request which does not have capabilities to remove or create the user. This is used to pass back to OBC request. For Provision() call, we may need to create a bucket with these credentials otherwise we just give back details in OB response and delete the bucket in Delete()

Good point, we should use we admin user instead. Will look deeper.

Reply:
@leseb: My first question was there is `adminOpsClient        *admin.API` part of `context` of `cephobjectstore`. This getting initialised in `checkObjectStoreHealth()`. Instead of doing that, can we  intialise that on `reconcileCreateObjectStore()` and use that `adminOpsClient` from `cephobjectstorecontext` for all `OBC` and `cephobjectstoreusers`. You might need to intialise the `adminOpsClient` before  `checkObjectStoreHealth()` for addressing `dashboard` usecase.